lib/wrapThenable.js

```diff
@@ -5,28 +5,37 @@ const {
   kReplyHijacked
 } = require('./symbols')
 
-function wrapThenable (thenable, reply) {
+function wrapThenable (thenable, reply, channels, store) {
```

Comment:
I don't think this function should be modified, I would recommend to keep the change between `preHandlerCallback` and `reply.send()`.
Specifically, I would emit `asyncStart` and `asyncEnd` within `reply.send()`, as there are multiple ways to call that function (synchronous and asynchronous)